### ISSUE TITLE
[BUG] Docker setup script fails on sle-Micro 5.4

### ISSUE THREAD

**Issue by susesgartner** [rancher/rancher#43693]
**Rancher Server Setup**
- Rancher version: v1.5.0 (RKE1)
- Installation option: docker

**Information about the Cluster**
- Cluster Type: RKE1 downstream
Cluster node configuration: 1 node all roles
* Imported: running kubectl apply
 * Infrastructure Provider: vmware

**User Information**
Node user: root

**Describe the bug**
While attempting to setup a sle micro 5.4 node for use in a rke1 provisioned k8s cluster the docker install script fails with "ERROR: Unsupported distribution 'sle-micro'" 

**To Reproduce**
1. Install sle-micro 5.4 on a machine
2. Run curl https://releases.rancher.com/install-docker/20.10.sh | sh

**Result**
The script fails with: "ERROR: Unsupported distribution 'sle-micro'" 

**Expected Result**
Docker is setup properly.

**Screenshots**
<img width="605" alt="image" src="https://github.com/rancher/rancher/assets/136753565/e849855b-9a97-4f97-97a2-2ab6069ec146">

**Additional context**
We are trying to support sle micro 5.4 for rke1, rke2, k3s on vmware, azure and linode (only imported clusters currently). To do this we need to manually create a sle micro 5.4 cluster and import it into rancher. 

This issue is specific to RKE1

**Comment by snasovich** [comment]
@mitulshah-suse , please have your team look into updating our https://github.com/rancher/install-docker scripts to support Docker (let's start with 24.0.x) installation on SLE Micro 5.4? While I understand it's not officially supported by Docker (https://docs.docker.com/engine/install/#supported-platforms) but neither is SLES on x64 - and we maintain updated installation scripts ourselves. We should look into the same approach for SLE Micro. Also confirmed there is Docker 24.0.5 package available for SLE Micro 5.4, see https://scc.suse.com/packages/?name=SUSE%20Linux%20Enterprise%20Micro&version=5.4&arch=x86_64&query=docker&module=

**Comment by susesgartner** [comment]
@snasovich We did some further testing and the issue may not be limited just to the install script. 
**Setup**
1. Create a vm with SLE Micro 5.4
2. Manually install/start docker (tried 24.0.5 and 20.10.17) 
Note: I also tried setting up docker using zypper with the same result.
3. Setup passwordless ssh (tried RSA, and ed25519)
4. Create a user (optional, we also tried as root)
5. Add the user to the docker group
6. Try to "rke up" a single node cluster (tried from rke1 on the node and also from my local rke1 setup)

**Result**
The install fails with the following error when trying to run a docker command: 
`DEBU[0002] FIXME: Got an status-code for which error does not match any expected type!!!  error="error during connect: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info\": Unable to access the service on /var/run/docker.sock. The service might be still starting up. Error: ssh: rejected: connect failed (open failed)" module=api status_code=-1

WARN[0002] Failed to set up SSH tunneling for host [x.x.x.x]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": Unable to access the service on /var/run/docker.sock. The service might be still starting up. Error: ssh: rejected: connect failed (open failed)`

**Additional Notes**
1. I am able to login to the node from the rke1 setup and run docker commands/access the docker.sock.
2. I tried chmod 777 on the docker sock and tried again but encountered the same error.

**Comment by snasovich** [comment]
@susesgartner , thank you for this additional info. However, I recommend filing a separate issue for the fact `rke up` fails on SLE Micro with Docker already installed on it. This issue is specifically about making `install-docker` script work on SLE Micro. While I understand the ultimate goal is to ensure RKE1 provisioning works on SLE Micro, it's won't be manageable to handle all related issues as part of one umbrella issue.

**Comment by mitulshah-suse** [comment]
@snasovich Did further digging on this. There are a few challenges with getting this to work. 
Firstly to give context for SLES OS, the support is only limited to s390x for upstream docker. In general, AMIs come with docker preinstalled and we did not face any issues with node driver cluster installations on SLES.
Same is the case for SLE Micro.. we checked on AWS and azure, the images have docker preinstalled. So, in general we dont need to do anything with it.
Additionally, these machines are transactional servers.. and simply running `zypper install docker` does not work on them. We need to run them as `transactional-update` and that needs machine reboots for changes to take effect. 
I am not sure if we should be doing reboots as part of install docker script.
